packages/node/vro-language-server/src/server/request/collection/ServerCollection.ts

```diff
@@ -51,139 +55,120 @@ export class ServerCollection {
+                        })
+                    ))
+            )
+        )
+        this.logger.info("Modules and Actions collected from vRO")
+        return modules
     }
 
-    async installHintingPlugin() {
-        this.currentStatus.message = "Installing the plugin..."
-        this.logger.info(this.currentStatus.message)
+    collectModulesAndActions() {
+        this.getModulesAndActions().then(modules => {
+            this.hints.collectModulesAndActions(modules)
+        })
     }
 
-    async waitVroServices() {
-        this.currentStatus.message = "Waiting vRO services to start..."
-        this.logger.info(this.currentStatus.message)
+    collectObjects() {
+        this.getVroObjects().then(objects => {
+            this.hints.collectVroObjects(objects)
+        })
     }
 
-    async startCollecting() {
-        this.currentStatus.message = "Collecting..."
-        this.logger.info(this.currentStatus.message)
+    async getVroObjects() {
+        this.logger.info("Collecting vRO objects...")
 
+        let objects = {}
         try {
-            const outParams = await this.restClient.executeWorkflow("548056f3-7dad-49d8-93a4-ba4cff675b72")
-            const vpkResourceIdParam = !!outParams && outParams.length > 0 ? outParams[0] : undefined
-
-            if (
-                !vpkResourceIdParam ||
-                vpkResourceIdParam.name !== "vpkResourceId" ||
-                !vpkResourceIdParam.value ||
-                !vpkResourceIdParam.value.string ||
-                !vpkResourceIdParam.value.string.value
-            ) {
-                throw new Error("Missing vpkResourceId output parameter")
-            }
-
-            this.currentStatus.data.vpkResourceId = vpkResourceIdParam.value.string.value
+            objects = await this.restClient.getVroServerData()
         } catch (error) {
-            this.setError(error.message)
+            this.logger.error(error)
         }
-    }
-
-    async downloadHintsPack() {
-        this.currentStatus.message = "Downloading hints package..."
-        this.logger.info(this.currentStatus.message)
-
-        try {
-            const vpkFilePath = this.environment.resolveVpkFile()
-            const hintsDirPath = this.environment.resolveHintsDir()
-
-            if (!hintsDirPath) {
-                // should never happen
-                throw new Error("Undefined global hints dir path")
+        const plugins: HintPlugin[] = objects["plugins"]
+        const allObjects: vmw.pscoe.hints.IClass[] = []
+        const regex = new RegExp(/\/plugins\/([a-zA-Z0-9\_\-\.\/]+)/)
+
+        for (const plugin of plugins) {
+            const link = plugin.detailsLink.match(regex)
+            if (!link) {
+                throw new Error(`No plugin details found`)
             }
 
-            if (fs.existsSync(vpkFilePath)) {
-                fs.unlinkSync(vpkFilePath)
-            }
+            const parsedLink = link[0].substring(9).toString()
```

Comment:
can we expect cases with less links? count=0? And maybe add a comment what is specific for this link0?